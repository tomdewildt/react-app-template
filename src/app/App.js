import React from "react";
import { ThemeProvider } from "styled-components";
import { BrowserRouter } from "react-router-dom";

import Router from "../router/Router";
import theme, { GlobalStyle } from "../theme";

const App = () => (
    <ThemeProvider theme={ theme }>
        <GlobalStyle />
        <BrowserRouter>
            <Router />
        </BrowserRouter>
    </ThemeProvider>
);

export default App;
