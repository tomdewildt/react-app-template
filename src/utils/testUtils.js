/* eslint-disable import/no-extraneous-dependencies, react/prop-types */
import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import { expectSaga } from "redux-saga-test-plan";
import createStore from "redux-mock-store";

import defaultTheme from "../theme";

const defaultStore = createStore( [] );
const defaultState = { map: {} };

export const renderWithRouter = (
    component,
    entries = [ "/" ],
    state = defaultState,
    theme = defaultTheme,
) => {
    const store = defaultStore( state );
    return {
        store,
        ...render( component, {
            wrapper: ( { children } ) => (
                <ThemeProvider theme={ theme }>
                    <Provider store={ store }>
                        <MemoryRouter initialEntries={ entries }>
                            { children }
                        </MemoryRouter>
                    </Provider>
                </ThemeProvider>
            ),
        } ),
    };
};

export const renderWithProvider = (
    component,
    state = defaultState,
    theme = defaultTheme,
) => {
    const store = defaultStore( state );
    return {
        store,
        ...render( component, {
            wrapper: ( { children } ) => (
                <ThemeProvider theme={ theme }>
                    <Provider store={ store }>
                        { children }
                    </Provider>
                </ThemeProvider>
            ),
        } ),
    };
};

export const renderWithTheme = (
    component,
    theme = defaultTheme,
) => render( component, {
    wrapper: ( { children } ) => (
        <ThemeProvider theme={ theme }>
            { children }
        </ThemeProvider>
    ),
} );

export const runReducer = ( reducer, action, state ) => reducer( state, action );

export const runSaga = ( saga, ...args ) => expectSaga( saga, ...args );
