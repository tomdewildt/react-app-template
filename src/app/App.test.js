import React from "react";
import { axe } from "jest-axe";

import { renderWithProvider } from "../utils/testUtils";

import App from "./App";

describe( "Router", () => {
    it( "renders consistently", () => {
        const { asFragment } = renderWithProvider( <App /> );
        const fragment = asFragment();

        expect( fragment ).toMatchSnapshot();
    } );

    it( "should have no axe violations", async () => {
        const { container } = renderWithProvider( <App /> );
        const result = await axe( container );

        expect( result ).toHaveNoViolations();
    } );
} );
