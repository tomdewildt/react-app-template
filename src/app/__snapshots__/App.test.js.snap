// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders consistently 1`] = `
<DocumentFragment>
  <main
    class=""
    id="home"
  />
</DocumentFragment>
`;
